playbook/runner/agent.go

```diff
@@ -19,7 +20,32 @@ type ActionForAgent struct {
 	TemplateEnv actions.TemplateEnv      `json:"template_env"`
 }
 
-func GetActionForAgent(ctx context.Context, agent *models.Agent) (*ActionForAgent, error) {
+func GetActionForAgentWithWait(ctx context.Context, agent *models.Agent) (*ActionForAgent, error) {
+	action, err := getActionForAgent(ctx, agent)
+	if err != nil {
+		return nil, err
+	}
+
+	if action != nil {
+		return action, err
+	}
+
+	// Go into waiting state
+	select {
+	case <-time.After(LongpollTimeout):
+		return &ActionForAgent{}, nil
+
+	case <-ActionNotifyRouter.RegisterRoutes(agent.ID.String()):
```

Comment:
This would call RegisterRoutes for every action? Shouldn't their be 1 Register call, and then a GetChannelForAgent - Or rename to GetOrCreateChannelForAgent

Reply:
`RegisterRoutes` returns the existing channel if it was previously registered.

Reply:
Then it should probably be named GetOrCreateChannel